cli/src/torii/routing.rs

```diff
@@ -66,19 +67,23 @@ impl VerifiedQuery {
     pub fn validate(
         self,
         wsv: &mut WorldStateView,
-    ) -> Result<(ValidQueryRequest, PredicateBox), QueryExecutionFailure> {
-        let account_has_public_key = wsv.map_account(&self.payload.account_id, |account| {
-            account.signatories.contains(self.signature.public_key())
-        })?;
+    ) -> Result<(ValidQueryRequest, PredicateBox), ValidationFail> {
+        let account_has_public_key = wsv
+            .map_account(&self.payload.account_id, |account| {
+                account.signatories.contains(self.signature.public_key())
+            })
+            .map_err(QueryExecutionFailure::from)?;
         if !account_has_public_key {
             return Err(QueryExecutionFailure::Signature(String::from(
                 "Signature public key doesn't correspond to the account.",
-            )));
+            ))
+            .into());
         }
-        wsv.validator_view()
-            .clone() // Cloning validator is a cheap operation
-            .validate(wsv, &self.payload.account_id, self.payload.query.clone())
-            .map_err(|err| QueryExecutionFailure::Permission(err.to_string()))?;
+        wsv.validator_view().clone().validate(
```

Comment:
```suggestion
        wsv.validate(
```

isn't this a lot more sensible? I'm not asking you to make this change in this PR. I'm just wondering what you think

Reply:
I'm not sure. I don't like WSV being god-object. And by implementing `validate()` on it we'll make this types relations even harder.

And current `validator_view()` just returns a reference to the validor. So that `wsv` doesn't take the reponsibillity for functions provided by validator

Reply:
> And current `validator_view()` just returns a reference to the validor. So that wsv doesn't take the reponsibillity for functions provided by validator

`validator_view()` is always followed by `.clone()`. At least we can move the `clone()` into the `validator_view` function itself

@Erigara your thoughts?

Reply:
> > And current `validator_view()` just returns a reference to the validor. So that wsv doesn't take the reponsibillity for functions provided by validator
> 
> `validator_view()` is always followed by `.clone()`. At least we can move the `clone()` into the `validator_view` function itself
> 
> @Erigara your thoughts?

 Don't have strong opinion about this. But moving clone into the `validator_view` will be less repetative.
However i think we need to rename method in such case because there is no immutable view just clone.

Reply:
Created #3560